hip-0000-nft-music-metadata.md

```diff
@@ -0,0 +1,252 @@
+    "image": "ipfs://example-image-uri",
+    "creator": "MetaHeads",
+    "format": "audio/mpeg",
+    "type": "album",
+    "category": "music",
+    "properties": {
+      "artist": "MetaHeads",
+      "album": "Echoes of the Digital Age",
+      "creator_account": ["0.0.123456", "0.0.654321"],
+      "compilation": false,
+      "royalty_type": "account",
+      "sort_name": "MetaHeads",
+      "artist_type": "Group",
+      "duration": 720,
+      "upc": "123456789012",
+      "genre": {
+        "root": "Rock",
+        "sub": ["Electronic", "Alternative"]
+      },
+      "unlockables": {
+        "name": "Behind the Scenes Footage",
+        "description": "Exclusive backstage and studio footage of MetaHeads.",
+        "locked_content": {
+          "uri": "ipfs://example-locked-content-uri",
+          "type": "video/mp4"
+        },
+        "token_gated": ["12345", "67890"]
+      }
+    },
+    "files": [
+      {
+        "uri": "ipfs://track1-uri",
+        "type": "audio/mpeg",
+        "metadata": {
+          "name": "Digital Dreams",
+          "description": "The opening track setting the theme of the album.",
+          "properties": {
+            "streaming_royalties": 5,
+            "streaming_royalty_token": "0.0.988765",
+            "token_gated": true,
+            "track_number": 1,
+            "bpm": 120,
+            "duration": 240,
+            "key": "A Minor",
+            "location_created": "London, UK",
+            "recording_studio": "Digital Sound Studios",
+            "license": "CC BY-NC-ND 4.0",
+            "copyright": "MetaHeads, 2024",
+            "label": "Digital Rock Records",
+            "master": ["Digital Mastering Services"],
+            "publisher": ["MetaMusic Publishing"],
+            "pro": ["ASCAP"],
+            "release_date": "2024-01-01T00:00:00Z",
+            "isrc": "QMJMT2047001",
+            "lyrics": "[00:00:00] Echoes in the circuitry... [00:00:30] A digital awakening...",
+            "album": "Echoes of the Digital Age",
+            "upc": "123456789012",
+            "artist": "MetaHeads",
+            "genre": {
+              "root": "Rock",
+              "sub": ["Electronic"]
+            },
+            "credits": {
+              "producer": "Max Soundwave",
+              "mix": "Eddie Echo",
+              "composer": "MetaHeads",
+              "performers": ["Max Soundwave", "Eddie Echo", "Tina Tempo", "Bassline Ben"],
+              "songwriter": "MetaHeads"
+            }
+          }
+        }
+      },
+      {
+        "uri": "ipfs://track2-uri",
+        "type": "audio/mpeg",
+        "metadata": {
+          "name": "Circuit Breaker",
+          "description": "A powerful track about breaking free from digital constraints.",
+          "properties": {
+            // Similar metadata as for Track 1, with relevant changes for track number, ISRC, etc.
```

Comment:
I think that there is a music industry/standards based specific modification to this that I'd suggest:
1. Have the 1st entry in the `files` array represent the **album**.
2. This album would have `metadata.properties` that would specify all the common properties of the album plus all of its tracks.
3. Each of the subsequent entries in the `files` array represent **tracks**.
4. Each track would have `metadata.properties` that would only specify its different properties, in other words "with relevant changes for track number, ISRC, etc"
5. The client side applications who ingest/process/display the NFT metadata would be able to identify this album/ track distinction and "copy" the album properties that aren't specified across to each of the tracks (for all properties unspecified by the track)

However, how to make this backward compatible with the existing HIP-412 standard is a bit of a question mark - needs more thought.

Reply:
i really like this approach, but this standard is designed for backwards compatibility to HIP-412. essentially, we think of it as a sub-standard of HIP-412 if that makes sense...

Reply:
I've thought about this again, and I think there's a fairly straightforward way to implement this in a manner that is backwards compatible with HIP-412:

1. Add a new object at `token.properties.tracks`, which will have all the properties that are common to all tracks
2. In `token.files[].properties` only add the properties that are different between the tracks
3. Specify, as part of the HIP-920 standard (this HIP) the process through which client apps are expected to "merge" properties found in `token.properties.tracks` into `token.files[].properties`.

This way, (1) redundancy is much reduced, and (2) HIP-920 is still backwards compatible with HIP-412

Reply:
ok, i really dig this approach. cleans up the metadata a lot. in music you deal with so many redundancies.

interested to see what @ty-swirldslabs thinks as well

Reply:
hey @bguiz following up here as well. the group outlined that what you're describing is largely handled by the "album" object.